velox/functions/FunctionRegistry.h

```diff
@@ -95,4 +96,20 @@ resolveVectorFunctionWithMetadata(
 /// Clears the function registry.
 void clearFunctionRegistry();
 
+/// A function name is a companion function's if the name is an existing
+/// aggregation function name followed by one of specific suffixes.
+bool isCompanionFunctionName(
+    const std::string& name,
+    const std::unordered_map<std::string, exec::AggregateFunctionEntry>&
+        aggregateFunctions);
+
+/// The following helper functions return the list of scalar, aggregate, and
+/// window function names registered in Velox. The list can be sorted by setting
+/// the parameter sortNames to true.
+const std::vector<std::string> getScalarNames(bool sortResult = false);
```

Comment:
@aditi-pandit Aditi, would you share some context about the need for these APIs. I can see why we would need APIs to list scalar, aggregate and window functions along with signatures and metadata, but I don't understand why we need APIs to list function names (sorted).

Also, it is not clear what isCompanionFunctionName for? If we want to exclude companion functions from getAggregateNames, we should pass a flag to that API.

Also, it is not clear whether getWindowNames returns window and aggregate functions or just window functions.

+1 to other comments about not introducing dependency on velox_exec.

Reply:
@mbasmanova : 

Like you rightly suggested, the e2e usage is for the Presto Native side-car to get all function signatures and metadata.

To achieve this, one option is to use getFunctionSignatures, exec::aggregateFunctions and exec::windowFunctions and build the json metadata for the server. But those registries include companion functions which we don't want to expose at Presto. 

This current option is to get the (non-companion) Presto function names and look up the registries for the signatures and metadata. These functions to list the function names were already part of CoverageUtil and so were proposed in the registry for reuse.

Seems like you are suggesting that we could add a parameter to the getFunctionSignatures, exec::aggregateFunctions and exec::windowFunctions to filter companion functions. That's a good option as well.

getWindowNames is for non-aggregate window functions (the ranking and value functions).

@pramodsatya

Reply:
@aditi-pandit Aditi, thank you for clarifying. I suggest that we add flags to existing APIs to return list of functions w/ metadata and filter out companion functions.

Reply:
Thanks for the suggestion @mbasmanova, @aditi-pandit, updated the PR accordingly. Could you please take another look?